### ISSUE TITLE
Can you please add the Stanford dog dataset?

### ISSUE THREAD

**Issue by dgrnd4** [huggingface/datasets#4504]
## Adding a Dataset
- **Name:** *Stanford dog dataset*
- **Description:** *The dataset is about 120 classes for a total of 20.580 images. You can find the dataset here http://vision.stanford.edu/aditya86/ImageNetDogs/*
- **Paper:** *http://vision.stanford.edu/aditya86/ImageNetDogs/*
- **Data:** *[link to the Github repository or current dataset location](http://vision.stanford.edu/aditya86/ImageNetDogs/)*
- **Motivation:** *The dataset has been built using images and annotation from ImageNet for the task of fine-grained image categorization. It is useful for fine-grain purpose *

Instructions to add a new dataset can be found [here](https://github.com/huggingface/datasets/blob/master/ADD_NEW_DATASET.md).

**Comment by julien-c** [comment]
would you like to give it a try, @dgrnd4? (maybe with the help of the dataset author?)

**Comment by dgrnd4** [comment]
@julien-c i am sorry but I have no idea about how it works: can I add the dataset by myself, following "instructions to add a new dataset"?
Can I add a dataset even if it's not mine? (it's public in the link that I wrote on the post)

**Comment by benpicker** [comment]
@julien-c I'd be happy to try. I'm looking to do more contributions to become a better coder.

**Comment by mariosasko** [comment]
Hi! The [ADD NEW DATASET](https://github.com/huggingface/datasets/blob/master/ADD_NEW_DATASET.md) instructions are indeed the best place to start. It's also perfectly fine to add a dataset if it's public, even if it's not yours. Let me know if you need some additional pointers.

**Comment by khushmeeet** [comment]
If no one is working on this, I could take this up!

**Comment by dgrnd4** [comment]
@khushmeeet this is the link where I add the dataset already. If you can I would ask you to do this:
1) The dataset it's all in TRAINING SET: can you please divide it in Training,Test and Validation Set? If you can for each class, take the 80% for the Training set and the 10% for Test and 10% Validation
2) The images has different size, can you please resize all the images in 224,224,3? Look even at the last dimension ("3") because some images has dimension 4!

Thank you!!

**Comment by mariosasko** [comment]
Hi @khushmeeet! Thanks for the interest. You can self-assign the issue by commenting `#self-assign` on it. 

Also, I think we can skip @dgrnd4's steps as we try to avoid any custom processing on top of raw data. One can later copy the script and override `_post_process` in it to perform such processing on the generated dataset.

**Comment by khushmeeet** [comment]
Thanks @mariosasko 

@dgrnd4 As dataset is there on Hub, and preprocessing is not recommended. I am not sure if there is any other task to do. However, I can't seem to find relevant `.py` files for this dataset in GitHub repo.

**Comment by dgrnd4** [comment]
@khushmeeet @mariosasko The point is that the images must be processed and must have the same size in order to can be used for things for example "Training".

**Comment by mariosasko** [comment]
@dgrnd4 Yes, but this can be done after loading (`map` to resize images and `train_test_split` to create extra splits)

@khushmeeet The linked version is implemented as a no-code dataset and is generated directly from the ZIP archive, but our "GitHub" datasets (these are datasets without a user/org namespace on the Hub) need a generation script, and you can find one [here](https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/image_classification/stanford_dogs.py). `datasets` started as a fork of TFDS, so we share similar script structure, which makes it trivial to adapt it.

**Comment by dgrnd4** [comment]
@mariosasko in order to resize images I'm trying this method: 
```
for i in range(0,len(dataset['train'])): #len(dataset['train'])

  ex = dataset['train'][i] #i
  image = ex['image']
  image = image.convert("RGB") # <class 'PIL.Image.Image'> <PIL.Image.Image image mode=RGB size=500x333 at 0x7F84F1948150>
  image_resized = image.resize(size_to_resize) # <PIL.Image.Image image mode=RGB size=224x224 at 0x7F84F17885D0>

  dataset['train'][i]['image'] = image_resized 
```

Because the DatasetDict is formed by arrows that are immutable, the changing assignment in the last line of code, doesn't work!
Do you have any idea in order to get a valid result?

**Comment by khushmeeet** [comment]
#self-assign

**Comment by khushmeeet** [comment]
I have raised PR for adding stanford-dog dataset. I have not added any data preprocessing code. Only dataset generation script is there. Let me know any changes required, or anything to add to README.

**Comment by zutarich** [comment]
Is this issue still open, i am new to open source thus want to take this one as my start.

**Comment by mariosasko** [comment]
@zutarich This issue should have been closed since the dataset in question is available on the Hub [here](https://huggingface.co/datasets/dgrnd4/stanford_dog_dataset).

**Comment by AlanBlanchet** [comment]
I didn't know about this issue until now but i added my version of the dataset on the hub **with the bboxes** :
https://huggingface.co/datasets/Alanox/stanford-dogs

Hope this helps
